ha/util/conf_store.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+
+
+"""
+Wrappeprs for using the search APIs from conf store
+"""
+
+import json
+
+from ha.core.config.config_manager import ConfigManager
+from ha.k8s_setup.const import CLUSTER_CARDINALITY_KEY, CLUSTER_CARDINALITY_NUM_NODES, CLUSTER_CARDINALITY_LIST_NODES
+from cortx.utils.log import Log
+
+class ConftStoreSearch:
+
+    def __init__(self):
+        """Init method"""
+        self._confstore = ConfigManager.get_confstore()
```

Comment:
I think this needs to be in try catch. If self._confstore is None by any chance. Further logic will not work.

Reply:
This is already called from inside try catch in ha_setup. And everywhere static method is used so it will always be available after mini provisioning

Reply:
Ohkk..